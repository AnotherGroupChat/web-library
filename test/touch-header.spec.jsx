import React from 'react';
import renderer from 'react-test-renderer';

import TouchHeader from '../src/js/component/touch-header';
import collectionsFix from './fixtures/collections';

const flatten = (col, pathSoFar) => {
	pathSoFar.push(col);
	if('children' in col && col.children.length) {
		pathSoFar = flatten(col.children[0], pathSoFar);
	}
	return pathSoFar;
};

it('Renders touch header at arbitrary depth', () => {
	let path = flatten(collectionsFix[0], []);
	let tree = renderer.create(
		<TouchHeader path={ path } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Renders touch header at root', () => {
	let tree = renderer.create(
		<TouchHeader path={ [] } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});
