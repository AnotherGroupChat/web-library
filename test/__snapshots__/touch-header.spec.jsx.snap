// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders touch header at arbitrary depth 1`] = `
<header
  className="touch-header hidden-sm-up"
>
  <nav>
    <ul>
      <li
        className="forelast"
        data-id={4}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            depth-2
          </div>
        </div>
      </li>
      <li
        className="previous"
        data-id={5}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            depth-3
          </div>
        </div>
      </li>
      <li
        className="current"
        data-id={6}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            depth-4
          </div>
        </div>
      </li>
      <li
        className="next"
        data-id={7}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            
          </div>
        </div>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Renders touch header at root 1`] = `
<header
  className="touch-header hidden-sm-up"
>
  <nav>
    <ul>
      <li
        className="forelast"
        data-id={0}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            
          </div>
        </div>
      </li>
      <li
        className="previous"
        data-id={1}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            
          </div>
        </div>
      </li>
      <li
        className="current"
        data-id={2}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            /
          </div>
        </div>
      </li>
      <li
        className="next"
        data-id={3}
        tabIndex="0"
      >
        <div
          className="center-axis"
        >
          <div
            className="inner"
            onClick={[Function]}
          >
            
          </div>
        </div>
      </li>
    </ul>
  </nav>
</header>
`;
